torch.compile docs: "Profiling to understand torch.compile performance

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #102862

Docs on how to use torch.profiler.profile to understand torch.compile performance.

davidberard98 committed:
torch.compile docs: "Profiling to understand torch.compile performance

Docs on how to use torch.profiler.profile to understand torch.compile performance.

[ghstack-poisoned]

davidberard98 committed:
Update on "torch.compile docs: "Profiling to understand torch.compile performance"


Docs on how to use torch.profiler.profile to understand torch.compile performance.

[ghstack-poisoned]

davidberard98 committed:
Update on "torch.compile docs: "Profiling to understand torch.compile performance"


Docs on how to use torch.profiler.profile to understand torch.compile performance.

[ghstack-poisoned]

davidberard98 committed:
Update on "torch.compile docs: "Profiling to understand torch.compile performance"


Docs on how to use torch.profiler.profile to understand torch.compile performance.

[ghstack-poisoned]

davidberard98 committed:
Update on "torch.compile docs: "Profiling to understand torch.compile performance"


Docs on how to use torch.profiler.profile to understand torch.compile performance.

[ghstack-poisoned]

davidberard98 committed:
Update on "torch.compile docs: "Profiling to understand torch.compile performance"


Docs on how to use torch.profiler.profile to understand torch.compile performance.

[ghstack-poisoned]

davidberard98 committed:
Update on "torch.compile docs: "Profiling to understand torch.compile performance"


Docs on how to use torch.profiler.profile to understand torch.compile performance.

[ghstack-poisoned]

davidberard98 committed:
Update on "torch.compile docs: "Profiling to understand torch.compile performance"


Docs on how to use torch.profiler.profile to understand torch.compile performance.

[ghstack-poisoned]

eellison commented on docs/source/index.rst:
```diff
@@ -59,6 +59,7 @@ Features described in this documentation are classified by release status:
    compile/deep-dive
    compile/cudagraph_trees
    compile/performance-dashboard
+   compile/profiling_torch_compile
```
one small nit - want to link to this directly above the gpu profiling section ? note that the references should be fixed here: https://github.com/pytorch/pytorch/pull/103014/ and https://github.com/pytorch/pytorch/pull/102407... up to you

also when landing you can just wait for docs to build then merge -f

eellison commented on docs/source/compile/inductor_profiling.rst:
```diff
@@ -1,3 +1,5 @@
+.. _torchinductor-gpu-profiling:
```
This is a better fix (I guess this is how you are supposed to control how links show up ? ) but it was also fixed here: https://github.com/pytorch/pytorch/pull/103014/

eellison replied:
actually nvm, this just so you could link it, not so it would show up different in side panel

eellison commented on docs/source/compile/profiling_torch_compile.rst:
```diff
@@ -0,0 +1,200 @@
+
+torch.profiler is helpful for understanding the performance of your program at a kernel-level granularity - for example, it can show graph breaks and GPU utilization at the level of the program. The data provided by the profiler can often help users understand where to investigate further to understand model performance.
+
+To understand kernel-level performance, other toosl exist. NVIDIA's ncu tool can be used, or :ref:`inductor's profiling tools <torchinductor-gpu-profiling>`.
```
nice!

eellison reviewed (approved):
Great!!!

davidberard98 committed:
Update on "torch.compile docs: "Profiling to understand torch.compile performance"


Docs on how to use torch.profiler.profile to understand torch.compile performance.

[ghstack-poisoned]

davidberard98 commented:
@pytorchbot merge

davidberard98 commented:
@pytorchbot merge -f "windows error is unrelated to these docs"
